dist/ramda.js

```diff
@@ -4440,6 +4440,43 @@
         }, [], fns);
     };
 
+    // The algorithm used to handle cyclic structures is heavily
+    // inspired by underscore.js' implementation of isEqual
+    var _eqDeep = function _eqDeep(a, b, stackA, stackB) {
+        var idx = stackA.length;
+        while (--idx >= 0) {
+            if (stackA[idx] === a) {
+                return stackB[idx] === b;
+            }
+        }
+        stackA.push(a);
+        stackB.push(b);
+        var result;
+        switch (type(a)) {
+        case 'Array':
```

Comment:
I don't think thats an important concern Scott but you can support both in the object case by doing something like

```js
var value = a.valueOf();
if (isObject(value)) {
   return value === b.valueOf;
}
```